### PR TITLE
verify user and dataset are allowed for query

### PR DESCRIPTION
**Pull request by PaulineMauryL**

### CODE REVIEW

File: server/lomas_server/routes_admin.py [hunk 1]
```diff
@@ -97,9 +102,17 @@ def get_dataset_metadata(
     """
     from app import app  # pylint: disable=C0415
 
+    dataset_name = query_json.dataset_name
+    if not app.state.admin_database.has_user_access_to_dataset(
```
Comment (damienbfs) [inline server/lomas_server/routes_admin.py:106]: Don't we already check at in admin_database.py at line 158 or is this different?

File: server/lomas_server/routes_admin.py [hunk 2]
```diff
@@ -143,6 +157,14 @@ def get_dummy_dataset(
...
```
Comment (damienbfs) [inline server/lomas_server/routes_admin.py:161]: Same comment as above

File: server/lomas_server/routes_admin.py (hunk 1 above)
Comment (PaulineMauryL) [reply server/lomas_server/routes_admin.py:106]: No for the dummy and metadata, until now we only checked that the dataset exists.
We did not verify who was querying and if they had access to the dataset. 
After the discussion https://github.com/dscc-admin-ch/lomas/issues/127 to query anything related to data (including 'public' information) the user is verified and must have access to the dataset. We could do more fine-grained authorization in the future.

File: server/lomas_server/routes_admin.py (hunk 2 above)
Comment (PaulineMauryL) [reply server/lomas_server/routes_admin.py:161]: same answer as above

File: server/lomas_server/routes_admin.py (hunk 1 above)
Comment (PaulineMauryL) [reply server/lomas_server/routes_admin.py:106]: just realised the lines number are different between branches and you are probably mentioning the decorator from the abstract class in master which is not in the mongodb or yaml child class.
so I guess it was a logic error that we missed from before but that we can re-use now and put in the child classes.

Comment (PaulineMauryL) [reply server/lomas_server/routes_admin.py:106]: I was starting to change but the function `get_dataset_metadata` is also used in part where it does not make sense to check the user (like private_dataset_factory) 🤔 
Maybe we could have 2 version of the method `get_dataset_metadata` (one with the check and one not) but then I am not sure it is better than what is currently in the PR.

Comment (damienbfs) [reply server/lomas_server/routes_admin.py:106]: Thanks for looking it up!
 
More generally, I guess it raises the question as to where we want to enforce authorization in the server.
- The decorator functions are neat, with the downside that we mix authorization and other functionality in the same code
- We could go down the road of completely separating both things (ie. similarly to what you are doing here, but maybe with a dedicated file/class/package for these type of checks) with the downside that it produces a bit more code?

I am not sure yet which option I would prefer.

Comment (PaulineMauryL) [reply server/lomas_server/routes_admin.py:106]: Yes I like the second option more. So far, the decorators were mainly oriented/thought towards functionnality (can not get a budget if user or dataset does not exist) but not really towards security. 
I think it is better to keep them separated, that way it is clearer and it avoids things like having to specify a user to get metadata within the server and so on.

**Review by damienbfs — APPROVED** [review]

**Pull request merged**